### ISSUE TITLE
Add Topical N-Grams for joint phrase and topic discovery

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#696]
## What you want and why

Add **Topical N-Grams** (Wang, McCallum & Wei, 2007): an LDA extension that jointly discovers topics and topic-specific multiword expressions. Unlike preprocessing-only phrase detection, it models whether adjacent tokens continue a phrase *conditional on the topic*, allowing the same word to participate in different expressions in different topical contexts.

This would strengthen the interpretability/measurement side of topica without treating phrase detection as merely a vocabulary-cleaning step. It is especially useful for policy, scientific, legal, news, and review corpora where concepts are frequently expressed as collocations.

Reference: Xuerui Wang, Andrew McCallum, and Xing Wei (2007), *Topical N-Grams: Phrase and Topic Discovery, with an Application to Information Retrieval*, ICDM. https://doi.org/10.1109/ICDM.2007.86

## Proposed surface

```python
model = topica.TopicalNGrams(num_topics=20, seed=13)
model.fit(docs, iters=1000)

model.top_words(10)
model.top_phrases(10)
model.top_phrases(topic=3, n=20)
```

Retain the common topica surface (`topic_word`, `doc_topic`, diagnostics, save/load), while adding a clear phrase representation and phrase-ranking API.

## Acceptance criteria

- Implements the canonical topic-conditioned n-gram continuation model (at least bigrams; document any chosen maximum order).
- Seed-reproducible fitting and a consistent `num_topics` / `iters` API.
- `top_phrases()` returns phrases with weights/probabilities both globally and by topic.
- Topic summaries can show words and phrases without silently replacing one with the other.
- Representative-document tools and persistence work as for other count-based models.
- Tests cover phrase extraction on planted corpora, edge cases for short documents and punctuation/token boundaries, reproducibility, and an independent reference or paper-derived checks.
- Documentation distinguishes it from `learn_phrases()` / `apply_phrases()`: preprocessing constructs a fixed vocabulary before fitting; Topical N-Grams learns topic-specific phrase structure during fitting.

## Alternatives

Topica already supports phrase preprocessing, which remains the simpler and often preferable workflow. This model earns its complexity only when topic-specific collocations are substantively central, not simply to make topic labels look nicer.
